lib/glific/providers/gupshup/partner/partner_api.ex

```diff
@@ -0,0 +1,153 @@
+defmodule Glific.Providers.Gupshup.PartnerAPI do
+  @moduledoc """
+  A module to handle fetching tier related information like quality rating and app rating using partner API
+  """
+
+  alias Glific.{
+    Caches,
+    Partners.Saas
+  }
+
+  alias Plug.Conn.Query
+
+  use Tesla
+  plug(Tesla.Middleware.Logger)
+
+  plug(Tesla.Middleware.FormUrlencoded,
+    encode: &Query.encode/1
+  )
+
+  @partner_url "https://partner.gupshup.io/partner/account"
+  @app_url "https://partner.gupshup.io/partner/app/"
+  @global_organization_id 0
+
+  @doc """
+  Fetch the stripe tax rates
+  """
+  @spec gupshup_partner_api :: map()
+  def gupshup_partner_api,
+    do: Saas.gupshup_partner_api()
+
+  # fetches partner token
+  @spec get_partner_token :: {:ok, map()} | {:error, any}
+  defp get_partner_token do
+    credentials = gupshup_partner_api()
+
+    url = @partner_url <> "/login"
+
+    # Using Cachex.get instead of Caches.get as token expire after 24hrs and we dont want to refresh the cache
+    {:ok, partner_token} = Cachex.get(:glific_cache, {@global_organization_id, "partner_token"})
```

Comment:
@AkhileshNegi as I said earlier also, Please remove Cachex from here and add a new function in the Caches file, if you don't have a valid functionality. 

I will fix the fetch part in the later PRs.

Reply:
made requested changes